hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/schemas/V0490FileSchema.java

```diff
@@ -128,36 +135,67 @@ public Set<StateDefinition> statesToCreate(@NonNull final Configuration config)
-            createGenesisSoftwareUpdateFiles(bootstrapConfig, hederaConfig, filesConfig, files);
-        }
+        // No-op, genesis system files are created via dispatch during the genesis transaction
     }
 
     // ================================================================================================================
     // Creates and loads the Address Book into state
 
-    private void createGenesisAddressBookAndNodeDetails(
-            @NonNull final BootstrapConfig bootstrapConfig,
-            @NonNull final HederaConfig hederaConfig,
-            @NonNull final FilesConfig filesConfig,
-            @NonNull final WritableKVState<FileID, File> files,
-            @NonNull final NetworkInfo networkInfo) {
+    public void createGenesisAddressBookAndNodeDetails(@NonNull final GenesisContext genesisContext) {
+        requireNonNull(genesisContext);
+        final var networkInfo = genesisContext.networkInfo();
+        final var filesConfig = genesisContext.configuration().getConfigData(FilesConfig.class);
+        final var bootstrapConfig = genesisContext.configuration().getConfigData(BootstrapConfig.class);
 
-        logger.debug("Creating genesis address book and node details files");
+        // Create the master key that will own both of these special files
+        final var masterKey = KeyList.newBuilder()
+                .keys(Key.newBuilder()
+                        .ed25519(bootstrapConfig.genesisPublicKey())
+                        .build())
+                .build();
+
+        // Create the address book file
+        final var addressBookFileNum = filesConfig.addressBook();
+        genesisContext.dispatchCreation(
+                TransactionBody.newBuilder()
+                        .fileCreate(FileCreateTransactionBody.newBuilder()
+                                .contents(genesisAddressBook(networkInfo))
+                                .keys(masterKey)
+                                .expirationTime(maxLifetimeExpiry(genesisContext))
+                                .build())
+                        .build(),
+                addressBookFileNum);
+
+        // Create the node details for file 102,  their fields are different from 101, addressBook
+        final var nodeDetail = new ArrayList<NodeAddress>();
+        for (final var nodeInfo : networkInfo.addressBook()) {
+            nodeDetail.add(NodeAddress.newBuilder()
+                    .stake(nodeInfo.stake())
+                    .nodeAccountId(nodeInfo.accountId())
+                    .nodeId(nodeInfo.nodeId())
+                    .rsaPubKey(nodeInfo.hexEncodedPublicKey())
+                    // we really don't have grpc proxy name and port for now.Temporary values are set.
+                    // After Dynamic Address Book Phase 2 release, we will have the correct values. Then update here.
+                    .serviceEndpoint(ServiceEndpoint.newBuilder()
+                            .ipAddressV4(Bytes.wrap("1.0.0.0"))
```

Comment:
Probably have a constant that explains this ? Need not be done in same PR

Reply:
Need to ask DAB team myself. 🤔